### ISSUE TITLE
Improve RandomCast

### ISSUE THREAD

**Issue by Belle-PL** [google/error-prone#2414]
### Description of the problem / feature request:

> Hi, I have encountered three false negatives when Error Prone tried to detect Random Cast.

### Feature requests: what underlying problem are you trying to solve with this feature?

> Based on the description in the [official document](https://errorprone.info/bugpattern/RandomCast), Error Prone should have reported warnings to detect the bug because when right values are cast to integer, variable a will always be equal to zero.

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

```java
public void testRandomCast() {
   // should report warnings in the following three lines
   int a = (int)(Math.random());
   a = (int)(random.nextFloat());
   a = (int)(random.nextDouble());
}
```

### What version of Error Prone are you using?

> Version: 2.7.1

**Comment by Belle-PL** [comment]
@graememorgan Hi, Dr.Morgan, could you please kindly help me check this issue? Thanks!

**Comment by cushon** [comment]
Catching all possible occurrences of a particular bug pattern isn't a goal for Error Prone, e.g. the following example also involves a potentially buggy cast of the result of a call to `Math.random()`, but detecting it would require inter-procedural analysis that's outside Error Prone's capabilities:

```
  void f() {
    g(() -> Math.random());
  }

  int g(Supplier<Double> s) {
    double d = s.get();
    return (int) d;
  }
```

Our focus is mostly on detecting patterns with simple intraprocedural analysis.

But in this case it's easy enough to have the check skip the parens and recognize e.g. `(int)(Math.random())`
